pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -94,10 +91,17 @@ abstract class Analytics {
       toolsMessageVersion: kToolsMessageVersion,
       fs: fs,
       gaClient: gaClient,
-      surveyHandler: SurveyHandler(homeDirectory: homeDirectory, fs: fs),
+      surveyHandler: SurveyHandler(
+        dismissedSurveyFile: fs.file(p.join(
```

Comment:
I see this file path being constructed twice, here on line 95 and later on line 174, can we make it a shared static function on the `SurveyHandler` class that takes in the home path as an argument, that way if/when we change the path we only have to change one place? Or, if the path will always be the same relative to home, can we just pass in the home path to the constructor and have the constructor figure it out?

Reply:
Or we could create an instance of `Artifacts` and pass the filesystem and homedirectory in the constructor so we could simply run something like the below

```dart
import 'package:file/file.dart';
import 'package:file/memory.dart';
import 'package:path/path.dart' as p;

class Artifacts {
  final Directory homeDirectory;
  final FileSystem fs;

  Artifacts({required this.homeDirectory, required this.fs});

  File get dismissedSurveyFile =>
      fs.file(p.join(homeDirectory.path, 'path', 'to', 'dismissedSurvey.json'));

  File get logFile =>
      fs.file(p.join(homeDirectory.path, 'path', 'to', 'analytics.log'));
}

void main() {
  final fs = MemoryFileSystem.test();
  final homeDirectory = fs.directory('home');

  homeDirectory.createSync();

  // Create an instance that we can pass around
  final artifacts = Artifacts(homeDirectory: homeDirectory, fs: fs);

  final dismissedSurveyFile = artifacts.dismissedSurveyFile;
  dismissedSurveyFile.createSync(recursive: true);

  final logFile = artifacts.logFile;
  logFile.createSync(recursive: true);
}
```

But either way, I was thinking of adding this in a separate PR

Reply:
I think overall, I would try to avoid adding new classes unless you have a set of behaviors (functions) that:

1. mutate state that they exclusively own; and/or
2. you need to model a hierarchy of types (e.g. Animal, Dog, Cat)

Thus, I would prefer these just be top level functions

Reply:
If you decide to keep the dependency on package:file (which I think is reasonable), these functions could simply take one argument, `Directory homeDirectory`, and then you could specify the path to the desired files via the `FileSystemEntity.childDirectory` and `FileSystemEntity.childFile` methods, which will ensure the appropriate path separator is used for the file system your home directory was created from.

Reply:
Doing this in a separate PR SGTM.